### PR TITLE
Add challenge-authorization integration tests

### PR DESCRIPTION
**Pull request by agunde406**
This commit includes 4 new integration tests (and the updates required to write them):
1. 3-party circuit creation using challenge authorization
2. 2-party circuit creation using challenge authorization where the nodes are connection view endpoint before the circuit     proposal is submitted 
3. 2-party circuit creation using challenge authorization with restarts - currently ignored due to endpoint restart issue
4. Create two 2-party circuit between two nodes where the nodes both support 2 signing keys and the second circuit sets the second key as the public key for the member of the first node - currently ignored due to bugs within challenge authorization that must be fixed

### CODE REVIEW

File: splinterd/tests/admin/circuit_create.rs [hunk 1]
```diff
@@ -82,6 +85,266 @@ pub fn test_2_party_circuit_creation_challenge_authorization() {
...
+        .recv_timeout(timeout)
+        .expect("Unable to get new notifications");
+    assert_eq!(
+        notification,
+        TestEnum::Notification(PeerManagerNotification::Connected {
+            peer: PeerAuthorizationToken::from_public_key(
+                node_b
+                    .admin_signer()
+                    .public_key()
+                    .expect("Unable to get first node's public key")
```
Comment (isabeltomb) [inline splinterd/tests/admin/circuit_create.rs:145]: s/first/second

File: splinterd/tests/admin/circuit_create.rs [hunk 2]
```diff
@@ -82,6 +85,266 @@ pub fn test_2_party_circuit_creation_challenge_authorization() {
...
+        .next_event()
+        .expect("Unable to get next event");
+    assert_eq!(ready_a_event.event_type(), &EventType::CircuitReady);
+    assert_eq!(ready_b_event.event_type(), &EventType::CircuitReady);
+    assert_eq!(ready_a_event.proposal(), ready_b_event.proposal());
+
+    let circuit_a = node_a
+        .admin_service_client()
+        .fetch_circuit(&second_circuit_id)
+        .expect("Unable to get circuit from node_b")
```
Comment (isabeltomb) [inline splinterd/tests/admin/circuit_create.rs:335]: s/node_b/node_a

**Review by isabeltomb — DISMISSED** [review]

**Review by leebradley — CHANGES_REQUESTED** [review]

File: splinterd/tests/admin/circuit_create.rs [hunk 3]
```diff
@@ -82,6 +85,266 @@ pub fn test_2_party_circuit_creation_challenge_authorization() {
...
+///
+/// 1. Create and submit a `CircuitCreateRequest` from the first node
+/// 2. Wait until the proposal is available to the second node, using `list_proposals`
+/// 3. Verify the same proposal is available on each node
+/// 4. Submit the same `CircuitCreateRequest` created in the first step from the second node
+/// 5. Validate the duplicate proposal submitted in the previous step results in an error
+/// 6. Create and submit a `CircuitProposalVote` from the second node to accept the proposal
+/// 7. Wait until the circuit is available on the first node, using `list_circuits`
+/// 8. Verify the same circuit is available to each node
+/// 9. Rebeat steps 1-8 for a different circuit that sets the member public keys to the nodes
```
[the remark this answers is not in the document]
Comment (agunde406) [reply splinterd/tests/admin/circuit_create.rs]: Fixed

File: splinterd/tests/admin/circuit_create.rs [hunk 4]
```diff
@@ -82,6 +85,266 @@ pub fn test_2_party_circuit_creation_challenge_authorization() {
     shutdown!(network).expect("Unable to shutdown network");
 }
 
+/// Test that a 2-party circuit may be created on a 2-node network using challenge authorization
+/// after the nodes have already connected via endpoints.
+///
+/// 1. Add node_b as an unidentified peer on node_a
+/// 2. Wait for node_a to get connection notification of node_b.
```
[the remark this answers is not in the document]
Comment (agunde406) [reply splinterd/tests/admin/circuit_create.rs:88]: Simplified to remove the repeated steps.

**Review by leebradley — DISMISSED** [review]

File: splinterd/tests/framework/network.rs [hunk 5]
```diff
@@ -44,28 +45,38 @@ impl Network {
...
-            let signer = context.new_signer(context.new_random_private_key());
-            let public_key = signer
+            let admin_signer = context.new_signer(context.new_random_private_key());
+            let public_key = admin_signer
                 .public_key()
                 .map_err(|e| InternalError::from_source(Box::new(e)))?;
             let temp_dir = TempDir::new("scabbard_data")
                 .map_err(|e| InternalError::from_source(Box::new(e)))?;
 
+            let mut signers = vec![admin_signer.clone()];
```
Comment (peterschwarz) [inline splinterd/tests/framework/network.rs:63]: Is the admin signer intended to be used in the same way as the other signers?

Comment (agunde406) [reply splinterd/tests/framework/network.rs]: That was a hold over from the first challenge authorization test that was added. With the addition of supporting multiple keys, I updated the commit to remove the admin signer from the signers list.

**Review by peterschwarz — APPROVED** [review]

**Review by leebradley — APPROVED** [review]

**Pull request merged**